// @unibeautify/beautifier-prettier/dist/src/options.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const commonOptions = {
    insertPragma: "pragma_insert",
    printWidth: "wrap_line_length",
    requirePragma: "pragma_require",
    tabWidth: "indent_size",
    useTabs: [
        ["indent_style"],
        (options) => {
            if (options.indent_style === "tab") {
                return true;
            }
            return false;
        },
    ],
};
const markdownOptions = Object.assign({}, commonOptions, { proseWrap: "wrap_prose" });
delete markdownOptions.tabWidth;
delete markdownOptions.useTabs;
const scriptOptions = Object.assign({}, commonOptions, { arrowParens: [
        ["arrow_parens"],
        (options) => {
            switch (options.arrow_parens) {
                case "always":
                    return "always";
                case "as-needed":
                    return "avoid";
                default:
                    return "avoid";
            }
        },
    ], bracketSpacing: "object_curly_spacing", jsxBracketSameLine: "jsx_brackets", trailingComma: [
        ["end_with_comma"],
        (options) => {
            switch (options.end_with_comma) {
                case true:
                    return "es5";
                case false:
                    return "none";
                default:
                    return undefined;
            }
        },
    ], semi: "end_with_semicolon", singleQuote: [
        ["quotes"],
        (options) => {
            switch (options.quotes) {
                case "double":
                    return false;
                case "single":
                    return true;
                default:
                    return undefined;
            }
        },
    ] });
const jsonOptions = Object.assign({}, commonOptions);
delete jsonOptions.insertPragma;
delete jsonOptions.requirePragma;
delete jsonOptions.printWidth;
const vueOptions = Object.assign({}, commonOptions);
delete vueOptions.printWidth;
const options = {
    JSON: jsonOptions,
    Markup: commonOptions,
    Markdown: markdownOptions,
    Script: scriptOptions,
    Style: commonOptions,
    GraphQL: commonOptions,
    Vue: vueOptions,
};
exports.default = options;
